internal/controller/cilium_flow_collector.go

```diff
@@ -48,25 +59,59 @@ func discoverCiliumHubbleRelayAddress(ctx context.Context, ciliumNamespace strin
 	return address, nil
 }
 
-// newCiliumCollector connects to Ciilium Hubble Relay, sets up an Observer client, and returns a new Collector using it.
+// newCiliumFlowCollector connects to Cilium Hubble Relay, sets up an Observer client, and returns a new Collector using it.
 func newCiliumFlowCollector(ctx context.Context, logger *zap.Logger, ciliumNamespace string) (*CiliumFlowCollector, error) {
-	config, err := NewClientSet()
+	clientset, err := NewClientSet()
 	if err != nil {
+		logger.Error("Failed to create new client set", zap.Error(err))
 		return nil, fmt.Errorf("failed to create new client set: %w", err)
```

Comment:
Why not just always return a proper error and log at the caller instead of ignoring error message there?
https://github.com/illumio/cloud-operator/blob/54609d8bfa90d863175a66013aeebf741e340f65/internal/controller/streams.go#L395-L398
```suggestion
		return nil, fmt.Errorf("failed to create new client set: %w", err)
```